privacyidea/api/lib/prepolicy.py

```diff
@@ -2233,6 +2235,12 @@ def require_description(request=None, action=None):
     token_types = list(action_values.keys())
     type_value = request.all_data.get("type") or 'hotp'
     if type_value in token_types:
-        if not request.all_data.get("description"):
+        tok = None
+        serial = getParam(params, "serial")
+        if serial:
+            tok = get_one_token(serial=serial, rollout_state=ROLLOUTSTATE.VERIFYPENDING, silent_fail=True) or \
+                  get_one_token(serial=serial, rollout_state=ROLLOUTSTATE.CLIENTWAIT, silent_fail=True)
```

Comment:
I think it would be more pythonic to use:
```
if serial:
    for rs in (ROLLOUTSTATE.VERIFYPENDING, ROLLOUTSTATE.CLIENTWAIT):
        try:
            get_one_token(serial=serial, rollout_state=rs)
        except ResourceNotFoundError as _e:
            pass
```
To avoid multiple database request we could add an else block:
```
...
try:
    tok = get_one_token(serial=serial, rollout_state=rs)
except ResourceNotFoundError as _e:
    pass
else:
    break
```
This way we avoid augmenting the `get_one_token()` function.

Reply:
`get_one_token` does not allow a or list of states. It will only filter for *one* state.
This means, we need two calls!

I first started with fetching the exceptions. But as we need two calls, this is evil.

Yes, we need two database calls but we **only** need them if

- we are doing 2step enrollment in the 2nd step!
- and if it is not a verify-enrollment but a client-wait enrollment.

So I would like to stick with the current way in the PR.

Reply:
Ok, the complete code would be:
```
if serial:
    for rs in (ROLLOUTSTATE.VERIFYPENDING, ROLLOUTSTATE.CLIENTWAIT):
        try:
            get_one_token(serial=serial, rollout_state=rs)
        except ResourceNotFoundError as _e:
            pass
        else:
            break
```
So both states are covered and if a token is found with the state `ROLLOUTSTATE.VERIFYPENDING` the other isn't tested anymore.

Reply:
I prefer my one line

Reply:
This makes it also fucking complicated, since you also need to test for the existence of the description.